### PR TITLE
Half precision

### PR DESCRIPTION
**Pull request by Old-Shatterhand**
In case of classification with the BinaryCrossEntropy Loss, using sigmoid and BCE is the same as [BCE with logits](https://pytorch.org/docs/stable/generated/torch.nn.BCEWithLogitsLoss.html). I adapted the models (also regression) accordingly as BCE+L works with half precision, normal BCE doesn't.

### CODE REVIEW

**Review by ilsenatorov — APPROVED** [review]
I think the baseline should be a good thing when working with many different datasets.
